mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageIdManager.java

```diff
@@ -0,0 +1,301 @@
+        return createMessage(Optional.fromNullable(internalDate).or(new Date()),
+                    messageInformation.getSize(),
+                    messageInformation.getBodyStartOctet(),
+                    messageInformation.getContentIn(),
+                    FlagsOperations.prepareForAppend(flagsToBeSet, true, permanentFlagsProvider.get(mailboxSession)),
+                    messageInformation.getPropertyBuilder(),
+                    messageInformation.getAttachments(),
+                    mailboxId);
+    }
+
+    /**
+     * Create a new {@link MailboxMessage} for the given data
+     */
+    protected MailboxMessage createMessage(Date internalDate, int size, int bodyStartOctet, SharedInputStream content, Flags flags, PropertyBuilder propertyBuilder, List<MessageAttachment> attachments, MailboxId mailboxId) throws MailboxException {
+        return new SimpleMailboxMessage(messageIdFactory.generate(), internalDate, size, bodyStartOctet, content, flags, propertyBuilder, mailboxId, attachments);
+    }
+
+    private void dispatchAddedMessage(MailboxId mailboxId, MailboxSession mailboxSession, MessageMetaData messageMetaData) throws MailboxException {
+        MailboxMapper mailboxMapper = mailboxSessionMapperFactory.getMailboxMapper(mailboxSession);
+        SortedMap<MessageUid, MessageMetaData> map = new TreeMap<MessageUid, MessageMetaData>();
+        map.put(messageMetaData.getUid(), messageMetaData);
+        dispatcher.added(mailboxSession, map, mailboxMapper.findMailboxById(mailboxId));
+    }
+
+    private Function<MailboxMessage, MessageResult> createMessageToResult(final MessageResult.FetchGroup minimal) {
+        return new Function<MailboxMessage, MessageResult>() {
+            @Override
+            public MessageResult apply(MailboxMessage input) {
+                try {
+                    return ResultUtils.loadMessageResult(input, minimal);
+                } catch (MailboxException e) {
+                    throw Throwables.propagate(e);
+                }
+            }
+        };
+    }
+
+    private Predicate<MailboxMessage> createIsInMailboxesPredicate(final List<MailboxId> mailboxIds) {
+        return new Predicate<MailboxMessage>() {
+            @Override
+            public boolean apply(MailboxMessage mailboxMessage) {
+                return mailboxIds.contains(mailboxMessage.getMailboxId());
+            }
+        };
+    }
+
+    private Predicate<MailboxId> notIn(final List<MailboxId> alreadyInMailboxes) {
+        return new Predicate<MailboxId>() {
+            @Override
+            public boolean apply(MailboxId input) {
+                return !alreadyInMailboxes.contains(input);
+            }
+        };
+    }
+
+    private RuntimeException unwrap(RuntimeException runtime) throws MailboxException {
```

Comment:
please use a local specific exception for that, using RuntimeException is not a good idea